File: src/main/Java/boj_step16/boj11866.java
import java.io.*;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class boj11866 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int k = Integer.parseInt(tokenizer.nextToken());
        writer.append(josephus(n, k)).close();
    }
    private static String josephus(int n ,int k) {
        Queue<Integer> list = new LinkedList<>();
        StringBuilder temp = new StringBuilder("<");
        for (int i = 1; i <= n; i++) list.offer(i);
        while (list.size() != 1) {
            for (int i = 1; i < k; i++) list.offer(list.poll());
            temp.append(list.poll() + ", ");
        }
        temp.append(list.poll() + ">");
        return temp.toString();
    }
}

File: src/main/Kotlin/boj_step16/boj11866.kt

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (n, k) = r.readLine().split(" ").map { it.toInt() }
    val josephus = josephus(n, k)
    w.write("<")
    josephus.forEach {
        if (it == josephus.last()) w.write("$it>")
        else w.write("$it, ")
    }
    w.close()
}

fun josephus(n: Int, k: Int): LinkedList<Int> {
    val josephus = LinkedList<Int>()
    val queue = LinkedList<Int>()
    val j = 1
    for (i in 1 .. n) queue.add(i)
    while (queue.size != 1) {
        for (i in 1 until k) {
            queue.add(queue.removeFirst())
        }
        josephus.add(queue.removeFirst())
    }
    josephus.add(queue.removeFirst())
    return josephus
}